mobile_packet_verifier/src/burner.rs

```diff
@@ -33,43 +45,136 @@ where
+                    "Payer does not have enough balance to burn dcs"
+                );
                 continue;
             }
 
             tracing::info!(%total_dcs, %payer, "Burning DC");
-            if self.burn_data_credits(&payer, total_dcs).await.is_err() {
-                // We have failed to burn data credits:
-                metrics::counter!("burned", "payer" => payer.to_string(), "success" => "false")
-                    .increment(total_dcs);
-                continue;
+            let txn = self.solana.make_burn_transaction(&payer, total_dcs).await?;
+            match self.solana.submit_transaction(&txn).await {
+                Ok(()) => {
+                    handle_transaction_success(
+                        pool,
+                        payer,
+                        total_dcs,
+                        sessions,
+                        &self.valid_sessions,
+                    )
+                    .await?;
+                }
+                Err(err) => {
+                    let span = tracing::info_span!(
+                        "txn_confirmation",
+                        signature = %txn.get_signature(),
+                        %payer,
+                        total_dcs,
+                        max_attempts = self.retry_attempts
+                    );
+
+                    tokio::spawn(
+                        self.transaction_confirmation_check(
+                            pool, err, txn, payer, total_dcs, sessions,
+                        )
+                        .instrument(span),
+                    );
+                }
             }
+        }
 
-            // We succesfully managed to burn data credits:
+        Ok(())
+    }
 
-            metrics::counter!("burned", "payer" => payer.to_string(), "success" => "true")
-                .increment(total_dcs);
+    fn transaction_confirmation_check(
+        &self,
+        pool: &Pool<Postgres>,
+        err: S::Error,
+        txn: S::Transaction,
+        payer: PublicKeyBinary,
+        total_dcs: u64,
+        sessions: Vec<pending_burns::DataTransferSession>,
+    ) -> impl Future<Output = ()> {
+        let pool = pool.clone();
+        let solana = self.solana.clone();
+        let valid_sessions = self.valid_sessions.clone();
+        let retry_attempts = self.retry_attempts;
+        let check_interval = self.failed_check_interval;
 
-            // Delete from the data transfer session and write out to S3
-            pending_burns::delete_for_payer(pool, &payer, total_dcs).await?;
+        async move {
+            tracing::warn!(?err, "starting txn confirmation check");
+            // We don't know if the txn actually made it, maybe it did
 
-            for session in sessions {
-                self.valid_sessions
-                    .write(ValidDataTransferSession::from(session), &[])
-                    .await?;
+            let signature = txn.get_signature();
+            for check_idx in 0..retry_attempts {
```

Comment:
i chatted with @bbalser on this and we both agree this is a gap that cannot be solved readily without some additional structural changes. This PR is still an improvement on what was there previously.

Reply:
I think the next pass at this will most likely include some block height checking. That way we can be tell the difference between solana errors and the amount of time passing without our txn on being on chain.